tools/refresh.sh

```diff
@@ -259,7 +229,7 @@ for CONFIG in ${CONFIGS}; do
 
   # Show differences
 
-  if ! $CMPCONFIG $DEFCONFIG defconfig; then
+  if ! diff $DEFCONFIG defconfig; then
```

Comment:
Will this work on native windows or do we need to define a host variable for the diff command?

Reply:
diff should be available.  The native build uses the GNUWin32 which are an ancient port of GNU tools that run natively on Wondows.

Why not just unconditionally give the user the opportunity to save the defconfig if they provide the --prompt option even if there is not difference.  I like the behavior of a change like:

    -    if ! $CMPCONFIG $DEFCONFIG defconfig; then
    +    $CMPCONFIG $DEFCONFIG defconfig

Reply:
Ok, I will change to this behavour.

Reply:
```
====================================================================================
Configuration/Tool: arty_a7/nsh,CONFIG_RV32IM_TOOLCHAIN_GNU_RVGL
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Copy files
  Select CONFIG_HOST_LINUX=y
  Refreshing...
  Disabling CONFIG_RV32IM_TOOLCHAIN_GNU_RVGL
  Enabling CONFIG_RV32IM_TOOLCHAIN_GNU_RVGL
  Building NuttX...
  Normalize arty_a7/nsh
Saving the new configuration file
```
But the build log will show up "Saving the new configuration file" for each config. @patacongo do you like this?

Reply:
> diff should be available. The native build uses the GNUWin32 which are an ancient port of GNU tools that run natively on Wondows.

Just to confirm.  The DiffUtils package is included in GNUWin32:  http://gnuwin32.sourceforge.net/packages.html

Reply:
> But the build log will show up "Saving the new configuration file" for each config even there isn't any difference at all. @patacongo do you like this?

No.  The way that that worked in the past (and as it is still described in refrish.sh --help) is:

    ./refresh.sh is a tool for refreshing board configurations
    
    USAGE: ./refresh.sh [options] <board>:<config>+
    
    Where [options] include:
      --debug
         Enable script debug
      --silent
         Update board configuration without interaction.  Implies --defaults.
         Assumes no prompt for save.  Use --silent --prompt to prompt before saving.
      --prompt
         Prompt before updating and overwriting the defconfig file.  Default is to
         prompt unless --silent
      --defaults
         Do not prompt for new default selections; accept all recommended default values
      --nocopy
         Do not copy defconfig from nuttx/boards/<board>/configs to nuttx/.config
      --help
         Show this help message and exit
      <board>
         The board directory under nuttx/boards
      <config>
         The board configuration directory under nuttx/boards/<board>/configs
       Note: all configuration is refreshed if <board>:<config> equals all.

This means:

1. Normally there is a prompt unless `--defaults` is selected.  They it just automatically selects all of the default values.
2. If -`-silent` is selected then `--defaults` is automatically selected and there is no prompt.
3. If you select `--silent --prompt` then defaults are used and you are prompted.

So the prompt message should should not occur unless you want it.  The behavior you are described is a bug.  It is not a desirable because and does not follow the specified behavior of `refresh.sh`

Reply:
But, the prompt message is:
```
Save the new configuration (y/n)?
``
not:
```
Saving the new configuration file
```
The first one print only when --prompt is specified, the second one always print if there is any change.